src/compiler/checker.ts

```diff
@@ -5765,12 +5765,13 @@ namespace ts {
             return type.modifiersType;
         }
 
+        function isPartialMappedType(type: Type) {
+            return getObjectFlags(type) & ObjectFlags.Mapped && !!(<MappedType>type).declaration.questionToken;
+        }

```

Comment:
I'm not sure what you're getting at. The only way we can be certain that *all* properties of the type that results from a mapping operation are optional is if the mapped type includes a `?` in the template specification. That's what we're checking here.